usmqe_tests/ui/test_user_ui.py

```diff
@@ -0,0 +1,215 @@
+              "notifications_on": True
+              })
+    pytest.check(user.exists,
+                 "Check that edited user can be found on UI Users page")
+    pytest.check(user.email == "edited_email_for_{}@example.com".format(role),
+                 "Check that user's e-mail has been edited")
+    pytest.check(user.notifications_on,
+                 "Check that user's notification settings have been edited")
+    user_data["email"] = "edited_email_for_{}@example.com".format(role)
+    user_data["email_notifications"] = True
+    test.check_user(user_data)
+    """
+    :step:
+      Log in as the edited user
+    :result:
+      User can log in
+    """
+    url = urlparse(CONF.config["usmqe"]["web_url"])
+    app2 = Application(hostname=url.hostname,
+                       scheme=url.scheme,
+                       username=user.user_id,
+                       password=user.password)
+    ViaWebUI.navigate_to(app2.web_ui, "LoggedIn")
+    app2.web_ui.browser_manager.quit()
+
+    """
+    :step:
+      Delete user
+    :result:
+      User is deleted
+    """
+    user.delete()
+    pytest.check(not user.exists,
+                 "Check that user can't be found on UI Users page anymore.")
+
+
+@pytest.mark.testready
+@pytest.mark.author("ebondare@redhat.com")
+@pytest.mark.negative
+def test_user_creation_password_invalid(application, valid_session_credentials,
+                                        valid_normal_user_data, invalid_password):
+    """
+    Attempt to create a user with an invalid password.
+    """
+    """
+    :step:
+      Attempt to add user with invalid password.
+    :result:
+      User is not created.
+    """
+    user = application.collections.users.create(
+        user_id=valid_normal_user_data["username"],
+        name=valid_normal_user_data["name"],
+        email=valid_normal_user_data["email"],
+        notifications_on=valid_normal_user_data["email_notifications"],
+        password=invalid_password,
+        role=valid_normal_user_data["role"]
+    )
+    pytest.check(not user.exists,
+                 "Check that new user can't be found on UI Users page")
+    """
+    :step:
+      Check that user hasn't been created using API
+    :result:
+      There's no user with the specified username
+    """
+    test = tendrlapi_user.ApiUser(auth=valid_session_credentials)
+    user_data_password_invalid = copy.deepcopy(valid_normal_user_data)
+    user_data_password_invalid["password"] = invalid_password
+    asserts = {
+        "ok": False,
+        "reason": 'Not Found',
+        "status": 404}
+    not_found = test.get_user(user_data_password_invalid["username"], asserts_in=asserts)
+    pytest.check("Not found" in str(not_found),
+                 "Check that new user can't be found using API")
+
+
+@pytest.mark.testready
+@pytest.mark.author("ebondare@redhat.com")
+@pytest.mark.happypath
+def test_edit_email_password(valid_new_normal_user, valid_normal_user_data, valid_password):
```

Comment:
You can drop `valid_normal_user_data` fixture and use `valid_new_normal_user` directly.